examples/electron/package.json

```diff
@@ -1,7 +1,9 @@
 {
   "private": true,
   "name": "@theia/example-electron",
+  "productName": "Theia Electron Example",
   "version": "0.15.0",
+  "main": "src-gen/frontend/electron-main.js",
```

Comment:
It would break all downstreams? Maybe we are better to generate main.js file?

Reply:
By "breaking downstreams", do you mean that my changes to `@theia/cli` means that consuming applications have to specify a `main` field in their package.json files?

I could add a check that the package contains a `main` field here:
https://github.com/eclipse-theia/theia/blob/f5640d4454f6e44f1b1df508b51932ca3f789c85/dev-packages/application-manager/src/application-package-manager.ts#L80-L84

If I understand correctly, I don't think generating a different file will help - we need to pass the project directory to `electron` so it can pick up the package.json file.

Reply:
> breaks all downstreams

👍 Good point, on the hand, it is a dev thing only. I think it is uncommon to start an electron app from the sources with the `@theia/cli` unless you are a dev. We can add a `main`-check, as @mcgordonite suggested, if `main` is undefined in the `package.json` we do as before. Otherwise, we apply the new logic, and we fixed https://github.com/eclipse-theia/theia/issues/7172. Thoughts?

Reply:
Ok, sounds good, I thought electron has some fallback if main property is missing if not let’s do a check and fall back to all approach.

Reply:
Sounds good, I'll make those changes. Thanks!

Reply:
Electron fails to start if the main field is missing:
```bash
> electron .
Error launching app
Unable to find Electron app at /home/matt/theia/examples/electron                                                        
                                                                                                                                         
Cannot find module '/home/matt/theia/examples/electron'
```

I've added a fallback if the main field is not in the package.json.